File: dynamic/55/sol.py
from my import Solution

if __name__ == "__main__":
    solution = Solution()
    
    # Test case 1: Example from problem statement where jump is possible
    nums1 = [2,3,1,1,4]
    result1 = solution.canJump(nums1)
    print(f"Test case 1:")
    print(f"Input: {nums1}")
    print(f"Expected: True")
    print(f"Got: {result1}")
    assert result1 == True, "Test case 1 failed"
    
    # Test case 2: Example where jump is impossible
    nums2 = [3,2,1,0,4]
    result2 = solution.canJump(nums2)
    print(f"\nTest case 2:")
    print(f"Input: {nums2}")
    print(f"Expected: False")
    print(f"Got: {result2}")
    assert result2 == False, "Test case 2 failed"
    
    # Test case 3: Single element array
    nums3 = [0]
    result3 = solution.canJump(nums3)
    print(f"\nTest case 3:")
    print(f"Input: {nums3}")
    print(f"Expected: True")
    print(f"Got: {result3}")
    assert result3 == True, "Test case 3 failed"
    
    # Test case 4: Array with zero trapping
    nums5 = [1,0,1,0]
    result5 = solution.canJump(nums5)
    print(f"\nTest case 5:")
    print(f"Input: {nums5}")
    print(f"Expected: False")
    print(f"Got: {result5}")
    assert result5 == False, "Test case 4 failed"
    
    print("\nAll test cases passed!")

File: dynamic/55/sol.c
#include <stdio.h>
#include <stdbool.h>
#include <assert.h>

bool canJump(int* nums, int numsSize);

void runTest(int* nums, int size, bool expected, int testCase) {
    printf("Test case %d:\n", testCase);
    printf("Input: [");
    for (int i = 0; i < size; i++) {
        printf("%d%s", nums[i], i < size - 1 ? "," : "");
    }
    printf("]\n");
    printf("Expected: %s\n", expected ? "true" : "false");
    
    bool result = canJump(nums, size);
    printf("Got: %s\n", result ? "true" : "false");
    
    assert(result == expected);
    printf("Test case %d passed!\n\n", testCase);
}

int main() {
    // Test case 1: Jump possible
    int nums1[] = {2,3,1,1,4};
    runTest(nums1, 5, true, 1);
    
    // Test case 2: Jump impossible due to zero
    int nums2[] = {3,2,1,0,4};
    runTest(nums2, 5, false, 2);
    
    // Test case 3: Single element
    int nums3[] = {0};
    runTest(nums3, 1, true, 3);
    
    // Test case 4: Zeros except first
    int nums4[] = {2,0,0,0};
    runTest(nums4, 4, true, 4);
    
    // Test case 5: Zero trap
    int nums5[] = {1,0,1,0};
    runTest(nums5, 4, false, 5);
    
    printf("All test cases passed!\n");
    return 0;
}